src/arviz_plots/plots/utils.py

```diff
@@ -295,3 +295,113 @@ def add_reference_lines(
+    references : tuple, list or dict
+        Reference values to be plotted as bands/shaded regions.
+    orientation : str, default "vertical"
+        The orientation of the reference lines, either "vertical" or "horizontal".
+    aes_map : mapping of {str : sequence of str}, optional
+        Mapping of artists to aesthetics that should use their mapping in `plot_collection`
+        when plotted. Valid keys are the same as for `plot_kwargs`.
+
+        The default is to use an "overlay_band" aesthetic for all elements.
+
+        It is possible to request aesthetics without mappings defined in the
+        provided `plot_collection`. In those cases, a mapping of "ref_dim" to the requested
+        aesthetic will be automatically added.
+    plot_kwargs : mapping of {str : mapping or False}, optional
+        Valid keys are:
+
+        * "ref_band" -> passed to :func:`~arviz_plots.visuals.vspan` for vertical `orientation`
+          and to :func:`~arviz_plots.visuals.hspan` for horizontal `orientation`
+
+    sample_dims : list, optional
+        Dimensions to reduce unless mapped to an aesthetic.
+        Defaults to ``rcParams["data.sample_dims"]``
+    ref_dim : list, optional
+        Specifies the name of the reference dimension for reference values.
+    **kwargs : mapping of {str : sequence}, optional
+        Mapping of aesthetic keys to the values to be used in their mapping.
+        See :func:`~arviz_plots.PlotCollection.generate_aes_dt` for more details.
+
+    Returns
+    -------
+    plot_collection : PlotCollection
+        Plot collection with the reference lines added.
+
+    Examples
+    --------
+    Add reference bands from 0 to 2 and 3 to 5 for all variables.
+
+    .. plot::
+        :context: close-figs
+
+        >>> from arviz_plots import plot_dist, add_reference_bands, style
+        >>> style.use("arviz-variat")
+        >>> from arviz_base import load_arviz_data
+        >>> dt = load_arviz_data('centered_eight')
+        >>> pc = plot_dist(
+        >>>     dt,
+        >>>     kind="ecdf"
+        >>> )
+        >>> add_reference_bands(pc, references=[[0, 2], [3, 5]])
+    """
+    if plot_kwargs is None:
+        plot_kwargs = {}
+    if aes_map is None:
+        aes_map = {}
+    else:
+        aes_map = aes_map.copy()
+    aes_map["ref_band"] = aes_map.get("ref_band", ["overlay_band"])
+    if sample_dims is None:
+        sample_dims = list(set(plot_collection.data.dims).difference(plot_collection.facet_dims))
+    if isinstance(sample_dims, str):
+        sample_dims = [sample_dims]
+    if ref_dim is None:
+        ref_dim = ["ref_dim", "band_dim"]
+
+    plot_func = vspan if orientation == "vertical" else hspan
+
+    ref_ds = references_to_dataset(
+        references, plot_collection.data, sample_dims=sample_dims, ref_dim=ref_dim
+    )
+
+    requested_aes = set(aes_map["ref_band"]).difference(plot_collection.aes_set)
+    *ref_dim, _band_dim = ref_dim
```

Comment:
Not sure why pylint is not complaining, maybe the leading underscore. But I think we should do one of the following.

* use `*ref_dim, _` to not store the last dimension as anything
* use `*ref_dim, band_dim` and then check that `band_dim` has length 2 and raise an informative error otherwise (as I think the visual would break or behave weirdly in such case).

Reply:
Have added a check for this